### PR TITLE
Add support for AllowsAirRoutes building flag

### PR DESCRIPTION
**Pull request by KungCheops**
City connection is blocked if the city is blockaded by water and land.

### CODE REVIEW

**Review by LessRekkless — APPROVED** [review]

**Review by LessRekkless — CHANGES_REQUESTED** [review]
Actually, we [will] need an update the to City Connections entries in the Civilopedia [when airport gains AllowsAirRoutes]:
TXT_KEY_GOLD_TRADE_ROUTES_HEADING3_BODY
TXT_KEY_WORKERS_TRADEROUTES_HEADING3_BODY
TXT_KEY_HAPPINESS_CAUSESUNHAPPINESS_HEADING2_BODY

**Review by LessRekkless — APPROVED** [review]
Nevermind.  This is a problem for when a building actually gains AllowsAirRoutes.

File: CvGameCoreDLL_Expansion2/CvBuildingProductionAI.cpp [hunk 1]
```diff
@@ -389,6 +389,35 @@ int CvBuildingProductionAI::CheckBuildingBuildSanity(BuildingTypes eBuilding, in
 		}
 	}
 
+	if (pkBuildingInfo->AllowsAirRoutes())
+	{
+		CvCity* pCapital = kPlayer.getCapitalCity();
+		int iLocalBonus = 0;
+		if (m_pCity->IsRouteToCapitalConnected())
+		{
+			iLocalBonus -= 50;
```
[the remark this answers is not in the document]
Comment (azum4roll) [reply CvGameCoreDLL_Expansion2/CvBuildingProductionAI.cpp:398]: Surely this should be a positive instead if the city isn't connected, and 0 if it is?

[the remark this answers is not in the document]
Comment (KungCheops) [reply CvGameCoreDLL_Expansion2/CvBuildingProductionAI.cpp:398]: I copied this from the lighthouse logic. I agree that it's quite strange.

[the remark this answers is not in the document]
Comment (KungCheops) [reply CvGameCoreDLL_Expansion2/CvBuildingProductionAI.cpp:398]: But I didn't want building AI logic to be in the scope of this change so I just made a primitive version for now.

**Pull request merged**